File: android/app/src/main/java/com/izzatismail/midtrim/domain/usecase/CalculateTrimWindowUseCase.kt
package com.izzatismail.midtrim.domain.usecase

import com.izzatismail.midtrim.domain.entity.TrimWindow
import com.izzatismail.midtrim.domain.error.TrimWindowError

class CalculateTrimWindowUseCase {
    fun execute(videoDuration: Double, trimDuration: Double): TrimWindow {
        if (videoDuration < trimDuration) {
            throw TrimWindowError.VideoShorterThanTrimDuration(videoDuration, trimDuration)
        }

        val center = videoDuration / 2
        val halfTrim = trimDuration / 2
        val startTime = maxOf(0.0, center - halfTrim)
        val endTime = minOf(videoDuration, center + halfTrim)

        return TrimWindow(startTime = startTime, endTime = endTime)
    }
}

File: ios/MidTrim/MidTrim/Domain/UseCases/CalculateTrimWindowUseCase.swift
import Foundation

struct CalculateTrimWindowUseCase {
    func execute(videoDuration: Double, trimDuration: Double) throws -> TrimWindow {
        guard videoDuration >= trimDuration else {
            throw TrimWindowError.videoShorterThanTrimDuration(
                videoDuration: videoDuration,
                trimDuration: trimDuration
            )
        }

        let center = videoDuration / 2
        let halfTrim = trimDuration / 2
        let startTime = max(0, center - halfTrim)
        let endTime = min(videoDuration, center + halfTrim)

        return TrimWindow(startTime: startTime, endTime: endTime)
    }
}
